python/tests/test_nessie_cli.py

```diff
@@ -324,6 +337,62 @@ def test_merge() -> None:
     assert len(logs) == 1
 
 
+@pytest.mark.vcr
+def test_merge_json() -> None:
+    """Test merge operation."""
+    execute_cli_command(["branch", "dev"])
+    make_commit("merge.foo.bar", _new_table("test_merge"), "dev")
+    main_hash = ref_hash("main")
+    dev_hash = ref_hash("dev")
+
+    # Passing detached commit-id plus a _different_ hash-on-ref --> error
+    execute_cli_command(["merge", f"dev@{dev_hash}", "-c", main_hash, "-o", main_hash], ret_val=1)
+
+    merge_output = execute_cli_command(["--json", "merge", "dev", "-c", main_hash])
+    merge_response = MergeResponseSchema().loads(merge_output)
+
+    # Check merge response
+    assert_that(merge_response.target_branch).is_equal_to("main")
+    assert_that(merge_response.expected_hash).is_equal_to(main_hash)
+    assert_that(merge_response.effective_target_hash).is_equal_to(main_hash)
+    assert len(merge_response.source_commits) == 1
+    assert len(merge_response.details) == 1
+    assert_that(merge_response.details[0].source_commits[0]).is_equal_to(dev_hash)
+
+    logs = simplejson.loads(execute_cli_command(["--json", "log"]))
+    # we don't check for equality of hashes here because a merge
+    # produces a different commit hash on the target branch
+    assert len(logs) == 1
+    logs = simplejson.loads(execute_cli_command(["--json", "log", "dev"]))
+    assert len(logs) == 1
+
+
+@pytest.mark.vcr
+def test_merge_legacy() -> None:
+    """Test merge operation."""
+    execute_cli_command(["branch", "dev"])
+    make_commit("merge.foo.bar", _new_table("test_merge"), "dev")
+    main_hash = ref_hash("main")
+
+    merge_output = execute_cli_command(["merge", "dev", "-c", main_hash])
+    assert_that(merge_output).is_equal_to("Merge succeeded but legacy server did not respond with additional details.\n")
+
+
+@pytest.mark.vcr
+def test_merge_legacy_json() -> None:
+    """Test merge operation."""
+    execute_cli_command(["branch", "dev"])
+    make_commit("merge.foo.bar", _new_table("test_merge"), "dev")
+    main_hash = ref_hash("main")
+    dev_hash = ref_hash("dev")
+
+    # Passing detached commit-id plus a _different_ hash-on-ref --> error
+    execute_cli_command(["merge", f"dev@{dev_hash}", "-c", main_hash, "-o", main_hash], ret_val=1)
```

Comment:
is this part important for the legacy usecase?

maybe both `test_merge_legacy_json` and `test_merge_legacy` could just use a single vcr recording if we remove it, wdyt?